packages/shared-base/src/utils/compose.ts

```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+export function compose(init: React.ReactNode, ...f: ((children: React.ReactNode) => JSX.Element)[]) {
+    return f.reduceRight((prev, curr) => curr(prev), React.createElement('div', '', init))
```

Comment:
Your version is different than the original version. The original version creates a Fragment instead of a div.